feat: support testing on bun

achingbrain opened the pull request:
Run node tests with bun using env var `NODE_EXEC=bun`

achingbrain committed:
feat: support testing on bun

Run node tests with bun using env var `NODE_EXEC=bun`

achingbrain committed:
chore: ignore bun dep

achingbrain committed:
chore: skip bun test on windows

achingbrain commented:
Bah, it's broken on Windows, support might be added in a week or so - https://twitter.com/bunjavascript/status/1753267105909416079

SgtPooki reviewed (approved):
~good with this. Do we want to wait and see on bun+windows before merge?~

we should wait on this until bun+windows.. https://github.com/ipfs-examples/test-ipfs-example/actions/runs/7831891366/job/21369332451?pr=18

salieflewis commented:
> ~good with this. Do we want to wait and see on bun+windows before merge?~
> 
> we should wait on this until bun+windows.. https://github.com/ipfs-examples/test-ipfs-example/actions/runs/7831891366/job/21369332451?pr=18

Bun supports Windows 10 and later as of 1.1

https://bun.sh/blog/bun-v1.1

SgtPooki commented on test/bun.spec.ts:
```diff
@@ -0,0 +1,48 @@
+  this.timeout(540000)
+  this.slow(60000)
+
+  if (platform() === 'win32') {
+    it.skip('bun does not run on windows', () => {})
+
+    return
+  }
```
```suggestion
```

SgtPooki replied:
testing changes in https://github.com/ipfs-examples/test-ipfs-example/pull/19

SgtPooki committed:
deps: update bun & remove windows test skip (#19)

Removes the windows skip and updates bun

builds on top of https://github.com/ipfs-examples/test-ipfs-example/pull/18

achingbrain committed:
Merge branch 'main' into feat/support-testing-on-bun
